metricflow-semantics/metricflow_semantics/model/semantics/linkable_element_set.py

```diff
@@ -290,3 +288,104 @@ def only_unique_path_keys(self) -> LinkableElementSet:
                 if len(set(linkable_metrics)) <= 1
             },
         )
+
+    @property
+    @override
+    def derived_from_semantic_models(self) -> Sequence[SemanticModelReference]:
```

Comment:
Could we add some test cases for this and the filter by spec pattern operations? They're quite a bit more complex than the direct to spec conversion.

Reply:
Add some test cases.